packages/nextjs-demo/src/app/asr/page.tsx

```diff
@@ -0,0 +1,131 @@
+'use client';
+import '../globals.css';
+import React, { useState, useEffect } from 'react';
+import {
+  MicManager,
+  SpeechRecognitionFactory,
+  SpeechRecognitionBase,
+} from '@ai-jsx/lib/asr/asr';
+import { wordErrorRate } from 'word-error-rate';
+
+const HARVARD_SENTENCES_01_TRANSCRIPT = `Harvard list number one.
+     The birch canoe slid on the smooth planks.
+     Glue the sheet to the dark blue background.
+     It's easy to tell the depth of a well.
+     These days, a chicken leg is a rare dish.
+     Rice is often served in round bowls.
+     The juice of lemons makes fine punch.
+     The box was thrown beside the parked truck.
+     The hogs were fed chopped corn and garbage.
+     Four hours of steady work faced us.     
+     A large size in stockings is hard to sell.`;
+
+interface AsrComponentProps {
+  name: string;
+  id: string;
+  manager: MicManager | null;
+}
+
+const AsrComponent: React.FC<AsrComponentProps> = ({ name, id, manager }) => {
+  const [wer, setWer] = useState<number | null>(null);
+  const [latency, setLatency] = useState<number | null>(null);
+  const [recognizer, setRecognizer] = useState<SpeechRecognitionBase>(null);
+  const computeWer = (inText: string, refText: string) => {
+    const numLines = inText.split('\n').length - 1;
+    const refClean = refText
+      .split('\n')
+      .slice(0, numLines)
+      .join(' ')
+      .replace(/[^\w\s]|_/g, '')
+      .replace(/\s+/g, ' ')
+      .trim();
+    const inClean = inText
+      .replace(/[^\w\s]|_/g, '')
+      .replace(/\s+/g, ' ')
+      .trim();
+    return wordErrorRate(refClean, inClean);
+  };
+  const start = () => {
+    const recognizer = SpeechRecognitionFactory.create(id, manager);
+    setRecognizer(recognizer);
+    const el = document.getElementById(id) as HTMLTextAreaElement;
```

Comment:
Use `useRef` to get a handle to a specific element: https://react.dev/reference/react/useRef. (Perplexity can help you with this.)